CMakeLists.txt

```diff
@@ -1,3 +1,56 @@
-cmake_minimum_required(VERSION 3.11.0)
+cmake_minimum_required(VERSION 3.18.0)
 
-# TODO: Write proper build system :)
+project(Pizzas)
+
+
+include(FetchContent)
+FetchContent_Declare(
+    googletest
+    GIT_REPOSITORY https://github.com/google/googletest.git
+    GIT_TAG        main
+)
+FetchContent_MakeAvailable(googletest)
+
+
+find_library(PTHREAD_LIB pthread)
+
+set(SRC_LIST
+    src/Funghi.cpp
+    src/Margherita.cpp
+    src/Pizza.cpp
+    src/Pizzeria.cpp)
+
+set(FLAGS -Wall -Wextra -Werror -pedantic -Wconversion)
+
+# add sources as library
+add_library(${PROJECT_NAME}_LIB STATIC ${SRC_LIST})
+target_compile_options(${PROJECT_NAME}_LIB PRIVATE ${FLAGS})
+target_compile_features(${PROJECT_NAME}_LIB PRIVATE cxx_std_17)
+
+include_directories("${CMAKE_SOURCE_DIR}/src"
+                    "${CMAKE_SOURCE_DIR}/test/mocks")
+# main source code
+add_executable(${PROJECT_NAME} src/main.cpp)
+target_link_libraries(${PROJECT_NAME} ${PROJECT_NAME}_LIB ${PTHREAD_LIB})
+target_compile_options(${PROJECT_NAME} PRIVATE ${FLAGS})
+target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_17)
+
+# tests
+enable_testing()
+
+set(PROJECT_NAME_TEST ${PROJECT_NAME}_GT)
+
+# set(TEST_SRC_LIST test/PizzeriaTest.cpp)
+
+# add tests sources as library
+#add_library(${PROJECT_NAME_TEST}_LIB STATIC ${TEST_SRC_LIST})
+#target_compile_options(${PROJECT_NAME_TEST}_LIB PRIVATE ${FLAGS})
+#target_compile_features(${PROJECT_NAME_TEST}_LIB PRIVATE cxx_std_17)
+
+add_executable(${PROJECT_NAME_TEST} test/PizzeriaTest.cpp)
```

Comment:
Your solution is ok, but or checker requires test bin name to be `${PROJECT_NAME}-ut`